sentry/src/main/java/io/sentry/SentryEnvelopeItem.java

```diff
@@ -247,11 +248,15 @@ public CachedItem(final @Nullable Callable<byte[]> dataFactory) {
       this.dataFactory = dataFactory;
     }
 
-    public @Nullable byte[] getBytes() throws Exception {
-      if (bytes == null) {
+    public @NotNull byte[] getBytes() throws Exception {
+      if (bytes == null && dataFactory != null) {
         bytes = dataFactory.call();
       }
-      return bytes;
+      return orEmptyArray(bytes);
+    }
+
+    private static @NotNull byte[] orEmptyArray(final @Nullable byte[] bytes) {
```

Comment:
it does not need to be static

Reply:
Since it does not access the state, isn't it a good candidate for a static method?

Reply:
if access no state, doesn't override a base method, best to be static